lib/aws/msk/iam/sasl/signer/credential_resolver.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Aws::Msk::Iam::Sasl::Signer
+  class CredentialResolver
+    def from_credential_provider_chain(region)
+      client = Aws::Kafka::Client.new(region: region)
+      raise "No credentials found" unless client.config.credentials
+
+      client.config.credentials.credentials
```

Comment:
I'd recommend passing `client.config.credentials` as the credentials_provider in sigv4 so that credentials are fetched lazily.